src/components/plot/user-controls-grid.tsx

```diff
@@ -33,27 +37,123 @@ const UserControlsGrid = ({
 }: UserControlsGridProps): JSX.Element => {
   const [layout, setLayout] = useState<Layouts>({});
   const [gridProps, setGridProps] = useState<ResponsiveGridProps>({});
+  const [localControlData, setLocalControlData] = useState<any[]>(userControlData.userControlConfig);
+  const [validationComplete, setValidationComplete] = useState<boolean>(false);
 
-  const controlConfig = userControlData.userControlConfig;
+  const alertFunctions = useAlert();
 
   const defaultGridPropsConverted = convertKeysSnakeToCamel(defaultGridProps);
+  // since we're using strict mode, the useEffect is getting called twice in dev mode
+  // this is to guard against it
+  const setupStarted = useRef<boolean>(false);
 
-  // set layout and grid props for local react grid
+  /**
+  * It should be called once to validate the configuration data for the user controls
+  */
   useEffect(() => {
-    if (userControlData?.gridConfig?.layouts && Object.values(userControlData.gridConfig?.layouts).length > 0) {
+    if (setupStarted.current) return;
+    setupStarted.current = true;
+    if (localControlData?.length > 0) {
+
+      //Validate local controls for uid and type. Display error if any control doesn't have an uid or type
+      const invalidControlUid: string[] = [];
+      userControlData?.userControlConfig?.map((control: UserControlConfig) => {
+        if (!control?.uid) {
+          throw new ChaiseError('User Control Error', `Missing UID for the specified global user control  ${control?.label ? `labelled as "${control?.label}"` : ''}`);
```

Comment:
Same comment as above about using index if label is also undefined. This should be handled when the function is factored out.